src/ClassDiscovery/RelativeNamespaceDiscovery.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Robo\ClassDiscovery;
+
+use Robo\Common\ClassLoaderAwareTrait;
+use Robo\Contract\ClassLoaderAwareInterface;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Class RelativeNamespaceDiscovery
+ *
+ * @package Robo\Plugin\ClassDiscovery
+ */
+class RelativeNamespaceDiscovery extends AbstractClassDiscovery implements ClassLoaderAwareInterface
+{
+    use ClassLoaderAwareTrait;
+
+    /**
+     * @var string
+     */
+    protected $relativeNamespace = '';
+
+    /**
+     * @param string $relativeNamespace
+     *
+     * @return RelativeNamespaceDiscovery
+     */
+    public function setRelativeNamespace($relativeNamespace)
+    {
+        $this->relativeNamespace = $relativeNamespace;
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getClasses()
+    {
+        $classes = [];
+        $relativePath = $this->convertNamespaceToPath($this->relativeNamespace);
+
+        foreach ($this->getClassLoader()->getPrefixesPsr4() as $baseNamespace => $directories) {
+            $directories = array_filter(array_map(function ($directory) use ($relativePath) {
+                return $directory.$relativePath;
+            }, $directories), 'is_dir');
+
+            if ($directories) {
+                foreach ($this->search($directories, $this->searchPattern) as $file) {
+                    $relativePathName = $file->getRelativePathname();
+                    $classes[] = $baseNamespace.$this->convertPathToNamespace($relativePath.DIRECTORY_SEPARATOR.$relativePathName);
```

Comment:
This looks about right on inspection, but the Windows Appveyor tests are failing. You could potentially just use "/" here, I think, as DIRECTORY_SEPARATOR is not necessary in instances where paths are only being used by php apis.

Reply:
(Any change here will likely require changes in convertPathToNamespace & maybe elsewhere of course. I'm okay with any solution that passes on Appveyor.)